src/routes/safe/components/Settings/Advanced/index.tsx

```diff
@@ -0,0 +1,291 @@
+import { useSnackbar } from 'notistack'
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { generateColumns, MODULES_TABLE_ADDRESS_ID, getModuleData } from './dataFetcher'
+import { styles } from './style'
+
+import Identicon from 'src/components/Identicon'
+import Block from 'src/components/layout/Block'
+import Row from 'src/components/layout/Row'
+import { TableCell, TableRow } from 'src/components/layout/Table'
+import Table from 'src/components/Table'
+
+import { getGnosisSafeInstanceAt } from 'src/logic/contracts/safeContracts'
+import { TX_NOTIFICATION_TYPES } from 'src/logic/safe/transactions'
+import { grantedSelector } from 'src/routes/safe/container/selector'
+import createTransaction from 'src/routes/safe/store/actions/createTransaction'
+import {
+  safeParamAddressFromStateSelector,
+  safeNonceSelector,
+  safeModulesSelector,
+} from 'src/routes/safe/store/selectors'
+import styled from 'styled-components'
+import Modal from '../../../../../components/Modal'
+import Paragraph from '../../../../../components/layout/Paragraph'
+import IconButton from '@material-ui/core/IconButton'
+import Close from '@material-ui/icons/Close'
+import Hairline from '../../../../../components/layout/Hairline'
+import Col from '../../../../../components/layout/Col'
+import Link from '../../../../../components/layout/Link'
+import OpenInNew from '@material-ui/icons/OpenInNew'
+import { getEtherScanLink } from '../../../../../logic/wallets/getWeb3'
+import { md, secondary } from '../../../../../theme/variables'
+
+export const REMOVE_MODULE_BTN_TEST_ID = 'remove-module-btn'
+export const MODULES_ROW_TEST_ID = 'owners-row'
+
+const useStyles = makeStyles(styles)
+
+const AddressText = styled(Text)`
+  margin-left: 12px;
+`
+
+const InfoText = styled(Text)`
+  margin-top: 16px;
+`
+
+const Bold = styled.strong`
+  color: ${theme.colors.text};
+`
+
+const TableActionButton = styled(Button)`
+  background-color: transparent;
+
+  &:hover {
+    background-color: transparent;
+  }
+`
+
+const FooterWrapper = styled.div`
+  display: flex;
+  justify-content: space-around;
+`
+
+const openIconStyle = {
+  height: md,
+  color: secondary,
+}
+
+const Advanced: React.FC = () => {
+  const classes = useStyles()
+
+  const columns = generateColumns()
+  const autoColumns = columns.filter(({ custom }) => !custom)
+
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+  const nonce = useSelector(safeNonceSelector)
+  const granted = useSelector(grantedSelector)
+  const modules = useSelector(safeModulesSelector)
+  const moduleData = modules !== null ? getModuleData(List(modules)) : modules
+
+  const [viewRemoveModuleModal, setViewRemoveModuleModal] = React.useState(false)
+  const hideRemoveModuleModal = () => setViewRemoveModuleModal(false)
+
+  const [selectedModule, setSelectedModule] = React.useState(null)
+  const triggerRemoveSelectedModule = (module: [string, string]): void => {
```

Comment:
should this `module: [string, string]` type be extracted to a separate type? I've seen it's being used in a few places